File: user-service/src/main/java/rs/raf/user_service/repository/EmployeeRepository.java
package rs.raf.user_service.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import rs.raf.user_service.entity.Employee;

import java.util.Optional;

public interface EmployeeRepository extends JpaRepository<Employee, Long>, JpaSpecificationExecutor<Employee> {
    Optional<Employee> findByEmail(String email);
}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/repository/EmployeeRepository.kt
package pack.userservicekotlin.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.stereotype.Repository
import pack.userservicekotlin.domain.entities.Employee
import java.util.*

@Repository
interface EmployeeRepository :
    JpaRepository<Employee, Long>,
    JpaSpecificationExecutor<Employee> {
    fun existsByUsername(username: String): Boolean

    fun existsByEmail(email: String): Boolean

    fun findByEmail(email: String): Optional<Employee>

    fun findByJmbg(jmbg: String): Optional<Employee>
    fun findByUsername(username: String): Optional<Employee>
}
